### PR TITLE
Support for configurable signed entity types in aggregator

### LINKED ISSUE #1388: Implement configurable signed entity types in aggregator
## Why
We want to be able to activate the certification of some entity types only on some networks (e.g UTxO set activated on `testing-preview` but not on `release-mainnet`, `release-preprod` and `pre-release-preview`). This will allow us to work on entity types that are not ready for production with our monorepo.

## What
Add a configuration parameter on the aggregator that if activated provides the full list of entity types to use. If the parameter is not specified all the entity types are used.

## How

- [x] Adapt the aggregator state machine runner (`get_current_non_certified_open_message`) to compute the list based on the new configuration parameter (unit test the `compute_entity_types_to_use` function to create). In any case the `MithrilStakeDistribution` and `CardanoImmutableFiles` should always be signed.
- [x] ~~Add a warning message (asking to update) if the signer is not able to sign the entity type~~
- [x] Add configuration parameter to infrastructure 
- [x] Add configuration parameter to CI/CD
- [x] Update documentation with new configuration parameters

## Later
- Create configuration parameter in GitHub environment(s)
- Update the explorer to not display a tab for a signed entity type if no records exist

### PR DESCRIPTION
**Pull request by jpraynaud**
## Content
This PR includes a new optional `signed_entity_types` parameter to the aggregator configuration:
- If activated it provides the full list of entity types to use. 
- All the default entity types are used otherwise.

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [x] Update documentation website (if relevant)

## Issue(s)
Closes #1388

### CODE REVIEW

File: mithril-aggregator/src/runtime/runner.rs [hunk 1]
```diff
@@ -143,6 +144,42 @@ impl AggregatorRunner {
     pub fn new(dependencies: Arc<DependencyContainer>) -> Self {
         Self { dependencies }
     }
+
+    // Create the deduplicated list of allowed signed entity types.
+    // By default, the list contains the MithrilStakeDistribution and the CardanoImmutableFilesFull.
+    // The list can be extended with the configuration parameter `signed_entity_types`.
+    // The configuration parameter is a comma separated list of signed entity types.
+    // The signed entity types are defined in the [SignedEntityTypeDiscriminants] enum.
+    // The signed entity types are discarded if they are not declared in the [SignedEntityType] enum.
```
Comment (ghubertpalo) [inline mithril-aggregator/src/runtime/runner.rs:153]: Shouldn't the comments here be Rust doc?

File: mithril-common/src/entities/signed_entity_type.rs [hunk 2]
```diff
@@ -24,8 +24,9 @@ const ENTITY_TYPE_CARDANO_IMMUTABLE_FILES_FULL: usize = 2;
 /// are identified by their discriminant (i.e. index in the enum), thus the
 /// modification of this type should only ever consist of appending new
 /// variants.
-#[derive(Display, Debug, Clone, PartialEq, Eq, Serialize, Deserialize, EnumDiscriminants)]
+#[derive(Display, Debug, Clone, PartialEq, Eq, Serialize, Deserialize, EnumDiscriminants, Hash)]
```
Comment (dlachaumepalo) [inline mithril-common/src/entities/signed_entity_type.rs:27]: Is `Hash` needed?
```suggestion
#[derive(Display, Debug, Clone, PartialEq, Eq, Serialize, Deserialize, EnumDiscriminants)]
```

**Review by dlachaumepalo — APPROVED** [review]
LGTM 👍, just a comment.

File: mithril-test-lab/mithril-end-to-end/src/main.rs [hunk 3]
```diff
@@ -56,6 +56,10 @@ pub struct Args {
     #[clap(long, default_value = "thales")]
     mithril_era: String,
 
+    /// Signed entity types parameters (discriminants names in an ordered comma separated list).
+    #[clap(long)]
+    signed_entity_types: Option<String>,
```
Comment (ghubertpalo) [inline mithril-test-lab/mithril-end-to-end/src/main.rs:61]: It is possible to have Clap returning a Vec of Strings using [`value_delimiter`](https://docs.rs/clap/latest/clap/builder/struct.Arg.html#method.value_delimiter).

File: mithril-test-lab/mithril-end-to-end/src/mithril/infrastructure.rs [hunk 4]
```diff
@@ -23,12 +23,14 @@ pub struct MithrilInfrastructure {
 }
 
 impl MithrilInfrastructure {
+    #[allow(clippy::too_many_arguments)]
```
Comment (ghubertpalo) [inline mithril-test-lab/mithril-end-to-end/src/mithril/infrastructure.rs:26]: Should this be merged that way?

**Review by ghubertpalo — APPROVED** [review]

File: mithril-common/src/entities/signed_entity_type.rs (hunk 2 above)
Comment (jpraynaud) [reply mithril-common/src/entities/signed_entity_type.rs]: Good catch! Done :heavy_check_mark:

File: mithril-test-lab/mithril-end-to-end/src/main.rs (hunk 3 above)
Comment (jpraynaud) [reply mithril-test-lab/mithril-end-to-end/src/main.rs:61]: Done :heavy_check_mark:

File: mithril-test-lab/mithril-end-to-end/src/mithril/infrastructure.rs (hunk 4 above)
Comment (jpraynaud) [reply mithril-test-lab/mithril-end-to-end/src/mithril/infrastructure.rs]: Introduced a new config structure to avoid the clippy hack :+1:

**Pull request merged**